lib/browser/ipc-main-internal-utils.ts

```diff
@@ -1,9 +1,9 @@
-'use strict'
+import { ipcMainInternal } from '@electron/internal/browser/ipc-main-internal'
+import * as errorUtils from '@electron/internal/common/error-utils'
 
-const { ipcMainInternal } = require('@electron/internal/browser/ipc-main-internal')
-const errorUtils = require('@electron/internal/common/error-utils')
+type IPCHandler = (...args: any[]) => any
```

Comment:
could this potentially be more tightly typed with type parameters? idk if typescript has variadic type parameters

Reply:
The idea is the caller will provide the types.  Hence `handle` and `handleSync` have a generic

```js
<T extends IPCHandler> 
```